test/coverage/datamodel/UUriSerializerTest.cpp

```diff
@@ -23,16 +23,167 @@ class TestFixture : public testing::Test {
 
 	// Run once per execution of the test application.
 	// Used for setup of all tests. Has access to this instance.
-	TestFixture() = default;
-	~TestFixture() = default;
+	TestUUriSerializaerFixture() = default;
+	~TestUUriSerializaerFixture() = default;
 
 	// Run once per execution of the test application.
 	// Used only for global setup outside of tests.
 	static void SetUpTestSuite() {}
 	static void TearDownTestSuite() {}
 };
 
-// TODO replace
-TEST_F(TestFixture, SomeTestName) {}
+std::shared_ptr<uprotocol::v1::UUri> buildValidTestURI(
+    const std::string& authority = "192.168.1.10") {
+	std::shared_ptr<uprotocol::v1::UUri> uri =
+	    std::make_shared<uprotocol::v1::UUri>();
+	uri->set_authority_name(authority);
+	uri->set_ue_id(0x1100);
+	uri->set_ue_version_major(0xFF);
+	uri->set_resource_id(0x8000);
+	return uri;
+}
 
-}  // namespace
+// Positive test case - test serialization of UUri to string
+TEST_F(TestUUriSerializaerFixture, SerializeUUriToString) {
+	const std::shared_ptr<uprotocol::v1::UUri> testUUri = buildValidTestURI();
+	const std::string expectedUUri = "//192.168.1.10/1100/FF/8000";
+	const std::string actualUUri =
+	    uprotocol::datamodel::serializer::uri::AsString::serialize(*testUUri);
+	ASSERT_EQ(expectedUUri, actualUUri);
+}
+
+// Positive test case - test serialization of UUri with no authority to string
+TEST_F(TestUUriSerializaerFixture, SerializeUUriWithNoAuthorityToString) {
+	const std::shared_ptr<uprotocol::v1::UUri> testUUri = buildValidTestURI("");
+	const std::string expectedUUri = "/1100/FF/8000";
+	const std::string actualUUri =
+	    uprotocol::datamodel::serializer::uri::AsString::serialize(*testUUri);
+	ASSERT_EQ(expectedUUri, actualUUri);
+}
+
+// TODO: Enable this test once validator for UUri is implemented.
+// TEST_F(TestUUriSerializaerFixture, SerializeEmptyUUriToString) {
+// 	uprotocol::v1::UUri testUUri ;
+// 	std::string expectedUUri = "";
+// 	std::string actualUUri 	 =
+// uprotocol::datamodel::serializer::uri::AsString::serialize(testUUri);
+// 	ASSERT_EQ(expectedUUri, actualUUri);
+// }
+
+TEST_F(TestUUriSerializaerFixture, DeSerializeUUriStringWithScheme) {
+	const std::string uuriAsString = "up://192.168.1.10/1100/FF/8000";
```

Comment:
Is `up://` still a valid prefix on URIs? I thought the format was just `//` for non-local URIs

Reply:
UUri Specification - Figure 2 shows scheme may exist and a valid prefix on URI -> https://github.com/eclipse-uprotocol/up-spec/blob/main/basics/uri.adoc

I also checked steven implementation for java and he did check schema as well in java implementation:
https://github.com/eclipse-uprotocol/up-java/blob/fe4b560857695d71db82793921436f25b5d94136/src/main/java/org/eclipse/uprotocol/uri/serializer/UriSerializer.java